windows/code/RNDeviceInfoCPP.h

```diff
@@ -44,7 +44,29 @@ namespace winrt::RNDeviceInfoCPP
 
     bool isTabletHelper(std::string os)
     {
-      return std::regex_match(os, std::regex(".*windowsphone.*", std::regex_constants::icase));
+      // AnalyticsInfo is considered flakey or unreliable by most sources to be used, but no recommended alternatives were discovered.
+      // DeviceForm potential but not inclusive results:
+      // [Mobile, Tablet, Television, Car, Watch, VirtualReality, Desktop, Unknown]
+      // DeviceFamily potential but not inclusive results:
+      // [Windows.Desktop, Windows.Mobile, Windows.Xbox, Windows.Holographic, Windows.Team, Windows.IoT]
+      auto deviceForm = AnalyticsInfo::DeviceForm();
+      auto deviceFamily = AnalyticsInfo::VersionInfo().DeviceFamily();
+
+      bool isTabletByAnalytics = deviceForm == L"Tablet" || deviceForm == L"Mobile" || deviceFamily == L"Windows.Mobile";
+      bool isTabletByOsRegex = std::regex_match(os, std::regex(".*windowsphone.*", std::regex_constants::icase));
+
+      if (isTabletByAnalytics || isTabletByOsRegex) {
+          return true;
+      }
+
+      // If there is no keyboard, is that considered functionally equivalent to a 'tablet' in the Windows OS world?
```

Comment:
Mmm - that's a tough one. I think you're right *if* it isn't a `Handset` device type. No keyboard means it's definitely not a Desktop. Still might be a TV?

I think the difference between phone and tablet is screen size, it's used to determine layout style. Keyboard is used to determine interaction and should not be conflated I think except as some sort of backup

That reminds me that "Desktop" is actually a valid `getDeviceType` return type - added to support macOS in #1137 but not documented until just now. You could probably add that in this PR since it's so intimately related - "isTablet" should arguably just take do an internal 

```c++
// terrible untested typed in to a web text area code from a java dev follows
if (getDeviceType() == L"Tablet") { 
  return true; 
} 
return false;
```